### PR TITLE
Renamed the Insights project to conform to other WindowsAppSDK projects

### PR DESCRIPTION
**Pull request by ZenBird**
* Renamed the Insights lib project to conform to other WindowsAppSDK project names (Microsoft.WindowsAppSDK.Insights.lib) * Added a property in Native.C.Props to reference the library when linking against WindowsAppSDK.Foundation nuget package

### CODE REVIEW

File: build/AzurePipelinesTemplates/WindowsAppSDK-PublishTelemetry-Steps.yml [hunk 1]
```diff
@@ -32,9 +32,9 @@ steps:
   inputs:
     SourceFolder: '${{ parameters.WindowsAppSDKTelemetryBinariesDirectory }}\Release\$(buildPlatform)'
     Contents: |
-      WindowsAppSDKTelemetryResourceLib\WindowsAppSDKTelemetryResourceLib.lib
-      WindowsAppSDKTelemetryResourceLib\WindowsAppSDKTelemetryResourceLib.pdb
-    TargetFolder: '$(Build.ArtifactStagingDirectory)\WindowsAppSDKTelemetryResourceLib\lib\$(buildPlatform)'
+      Microsoft.WindowsAppSDKTelemetry\Microsoft.WindowsAppSDKTelemetry.lib
```
Comment (DrusTheAxe) [inline build/AzurePipelinesTemplates/WindowsAppSDK-PublishTelemetry-Steps.yml:35]: Usually it's Microsoft.WindowsAppSDK.Blah

**Comment by ZenBird** [comment]
> #ifndef __WINDOWSAPPSDKTELEMETRY_INCLUDED

Rename?

---
Refers to: dev/Common/WindowsAppSDKInsights.h:6 in 2060225. [](commit_id = 206022508a33d53463b9013dd37af27df954ff8e, deletion_comment = False)

File: dev/WindowsAppSDKInsightsResource/WindowsAppSDKInsightsResourceLib/WindowsAppSDKInsightsResourceLib.vcxproj [hunk 2]
```diff
@@ -41,6 +41,7 @@
     <ProjectGuid>{4243b8da-bdb5-4d65-b1d5-eeed375fad13}</ProjectGuid>
     <RootNamespace>WindowsAppSDKTelemetryResourceLib</RootNamespace>
```
Comment (ZenBird) [inline dev/WindowsAppSDKInsightsResource/WindowsAppSDKInsightsResourceLib/WindowsAppSDKInsightsResourceLib.vcxproj:42]: >WindowsAppSDKTelemetryResourceLib [](http://example.com/codeflow?start=19&length=33)

Rename

**Review by MythiliM — APPROVED** [review]

**Pull request merged**